compiler/rustc_lint/src/lib.rs

```diff
@@ -549,6 +548,7 @@ fn register_builtins(store: &mut LintStore) {
         "converted into hard error, see RFC #3535 \
          <https://rust-lang.github.io/rfcs/3535-constants-in-patterns.html> for more information",
     );
+    store.register_removed("box_pointers", "it was mostly historical and not particularly useful");
```

Comment:
```suggestion
    store.register_removed(
        "box_pointers", "it was mostly historical and does not detect other kinds of allocations"
    );
```
:blush:

Reply:
I followed the spirit of your suggestion, but ended up using a slightly different wording.